### PR TITLE
[v4.9-rhel] Backport of healthcheck timers

### PR DESCRIPTION
**Pull request by TomSweeneyRedHat**
Addresses: https://issues.redhat.com/browse/ACCELFIX-263,
https://issues.redhat.com/browse/RHEL-50506, and
https://issues.redhat.com/browse/RHEL-50507

Backports:  https://github.com/containers/podman/pull/22589 and
https://github.com/containers/podman/pull/22895 

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by TomSweeneyRedHat** [comment]
Thanks @Luap99 I'd a bit of merge mayhem with that particular file and it looks like I chose badly.  Will add those back.

**Comment by edsantiago** [comment]
@TomSweeneyRedHat I don't know what's broken here, but it looks real.

Hint: you can save yourself hours of CI time by running:
```console
$ hack/bats 220:'podman healthcheck$' 
```
(the dollar sign means "only tests that match 'podman healthcheck<ENDOFSTRING>', because all other tests in this file also include 'podman healthcheck' and there's no need to run them.)

File: test/system/220-healthcheck.bats [hunk 1]
```diff
@@ -34,14 +35,15 @@ function _check_health {
...
-    run_podman run -d --name healthcheck_c \
+    run_podman run -d --name healthcheck_c             \
                --health-cmd /healthcheck   \
-               --health-interval 1s        \
-               --health-retries 3          \
-               --health-on-failure=kill    \
+               --health-interval 1s                    \
+               --health-retries 3                      \
+               --health-on-failure=kill                \
+               --health-startup-cmd /home/podman/healthcheck \
```
Comment (edsantiago) [inline test/system/220-healthcheck.bats:43]: I think this is the bug. `$IMAGE` on this old branch is `20221018`, and that image does not include `/home/podman/healthcheck`.

Solutions:
1. bump TAG in helpers.bash to `20240123` (__may have unintended consequences!!__) or
2. remove this, I don't see health-startup-command being an important part of this test; or
3. dunno, I'm sure there are other options. Good luck.

File: test/system/220-healthcheck.bats [hunk 2]
```diff
@@ -88,7 +93,10 @@ Log[-1].Output   | \"Uh-oh on stdout!\\\nUh-oh on stderr!\"
 
     # Clean up
     run_podman rm -t 0 -f healthcheck_c
-    run_podman rmi   healthcheck_i
```
[the remark this answers is not in the document]
Comment (TomSweeneyRedHat) [reply test/system/220-healthcheck.bats:91]: @Luap99 I'm not sure what you comment is here?  I'm not seeing anything that I removed here, only added, or have I corrected that and GitHub is being brain dead?

[the remark this answers is not in the document]
Comment (TomSweeneyRedHat) [reply test/system/220-healthcheck.bats:91]: I see, It was removed earlier, but now I think it's correct, holler if not.

**Comment by TomSweeneyRedHat** [comment]
@edsantiago thanks for the pro-tip reminder on running with hack.  I had done that a long time ago, but completely forgot about it.  Will have to see if I can make a better note.

The change you suggested seems to have gotten it through the section that was failing, but now in my sandbox it looks to be failing later when the test is trying to force an error, but it's completing.  I'm hoping it's either my tired eyes missing it, or something messed up in my environment.  I've made your correction, and few nitty comment changes.  Fingers crossed.

File: test/system/220-healthcheck.bats (hunk 2 above)
[the remark this answers is not in the document]
Comment (Luap99) [reply test/system/220-healthcheck.bats:91]: you previous version was removing run_podman rmi healthcheck_i but that needed to go back with _build_health_check_image which you did so all good

**Review by Luap99 — APPROVED** [review]
LGTM, minikube tests are broken beyond repair.
We removed them on main already so I suggest we force merge this and then remove minkube tests from this branch in another PR or you just push this into here

**Comment by TomSweeneyRedHat** [comment]
@mheon care to push the shiny, but currently red, button on this one?  Please see @Luap99 's comments above

**Comment by mheon** [comment]
Done

**Pull request merged**